swarm_transport.go

```diff
@@ -48,7 +51,10 @@ func (s *Swarm) TransportForListening(a ma.Multiaddr) transport.Transport {
 	s.transports.RLock()
 	defer s.transports.RUnlock()
 	if len(s.transports.m) == 0 {
-		log.Error("you have no transports configured")
+		// make sure we're not just shutting down.
+		if s.transports.m != nil {
```

Comment:
Would be nice to actually model a swarm status with atomic ints. Inferring we're closing down based on nil vs. 0-length slice differentiation is a bit of a footgun.

Reply:
Note: this is a map, not a slice. We could have a second flag, but I'd prefer to avoid storing the state twice.